spec/controllers/annotations_controller_spec.rb

```diff
@@ -239,11 +256,133 @@
+        post_as user,
+                :update,
+                params: { content: 'New content!',
+                          id: annotation.id,
+                          result_id: other_grouping.current_result.id,
+                          assignment_id: assignment.id },
+                format: :js
+        assert_response :bad_request
+        expect(annotation.reload.annotation_text.content).to_not eq 'New content!'
+      end
+
+      it 'can destroy a deductive annotation' do
+        post_as user,
+                :destroy,
+                params: { id: annotation.id,
+                          result_id: result.id,
+                          assignment_id: assignment.id },
+                format: :js
+        expect(result.reload.annotations.size).to eq 0
+      end
+
+      it 'can destroy a deductive annotation when criteria assigned to graders' do
+        assignment.assignment_properties.update(assign_graders_to_criteria: true)
+        post_as user,
+                :destroy,
+                params: { id: annotation.id,
+                          result_id: result.id,
+                          assignment_id: assignment.id },
+                format: :js
+        expect(result.reload.annotations.size).to eq 0
+      end
+    end
   end
 
   describe 'an authenticated TA' do
-    let!(:user) { create(:admin) }
+    let!(:user) { create(:ta) }
     include_examples 'an authenticated admin or TA'
+
+    describe 'accessing annotations for results in an assignment with deductive annotations' do
+      let(:assignment) { create(:assignment_with_deductive_annotations) }
+      let(:result) { assignment.groupings.first.current_result }
+      let(:annotation) { result.annotations.first }
+      it 'cannot update a deductive annotation' do
+        post_as user,
+                :update,
+                params: { content: 'New content!',
+                          id: annotation.id,
+                          result_id: result.id,
+                          assignment_id: assignment.id },
+                format: :js
+        assert_response :bad_request
+        expect(annotation.reload.annotation_text.content).to_not eq 'New content!'
+      end
+
+      it 'cannot destroy a deductive annotation if unassigned to the annotation\'s criterion' do
+        other_criterion = create(:rubric_criterion, assignment: assignment)
+        assignment.assignment_properties.update(assign_graders_to_criteria: true)
+        create(:criterion_ta_association, criterion: other_criterion, ta: user)
+        post_as user,
+                :destroy,
+                params: { id: annotation.id,
+                          result_id: result.id,
+                          assignment_id: assignment.id },
+                format: :js
+        assert_response :bad_request
+        expect(result.reload.annotations.size).to eq 1
+      end
+
+      it 'cannot destroy a deductive annotation if assigned to a criterion of a different type '\
```

Comment:
I was thinking:
Similar to this test, but perhaps missing here, is a test to check if the TA cannot destroy a deductive annotation if assigned to a criterion of the same type and id as one for deductive annotations in an assignment, but for a different assignment. 

But I realized:
The same criterion type can never have the same ID regardless of assignment.

However:
Should there be a test for criteria across assignments?